scripts/mkstatic.py

```diff
@@ -1478,69 +1504,87 @@ def write(self, metadata_request, metadata_define, arguments, debug):
                                 # will catch any out of bound reads with the appropriate compiler flags. It naturally
                                 # deals with non-uniform block sizes etc.
                                 pass
-                            elif var.rank:
+                            # Some older versions of GNU currently in use can not do these variable size tests on strings
+                            # 0x5b6fdd gimplify_expr(tree_node**, gimple**, gimple**, bool (*)(tree_node*), int)
+                            #   /tmp/role.apps/spack-stage/spack-stage-gcc-9.2.0-ku6r4f5qa5obpfnqpa6pezhogxq6sp7h/spack-src/gcc/gimplify.c:13477
+                            elif var.rank and not var.type == 'character':
                                 assign_test = '''        ! Check if variable {var_name} is associated/allocated and has the correct size
         if (size({var_name}{dim_string})/={var_size_expected}) then
           write({ccpp_errmsg}, '(2(a,i8))') 'Detected size mismatch for variable {var_name}{dim_string} in group {group_name} before {subroutine_name}, expected ', &
                                            {var_size_expected}, ' but got ', size({var_name}{dim_string})
           ierr = 1
           return
         end if
-'''.format(var_name=local_vars[var_standard_name]['name'].replace(dim_string_target_name, ''), dim_string=dim_string, var_size_expected=var_size_expected,
+'''.format(var_name=local_vars[var_standard_name]['name'].replace(dim_string_target_name, ''),
+           dim_string=dim_string,
+           var_size_expected=var_size_expected,
            ccpp_errmsg=CCPP_INTERNAL_VARIABLES[CCPP_ERROR_MSG_VARIABLE], group_name = self.name,
            subroutine_name=subroutine_name)
                         # end if debug
 
                         # kind_string is used for automated unit conversions, i.e. foo_kind_phys
                         kind_string = '_' + local_vars[var_standard_name]['kind'] if local_vars[var_standard_name]['kind'] else ''
 
+                        # conditional is the conditional allocation, which can be '.true.', '.false.', or any regular Fortran logical expression
+                        conditional=conditionals[var_standard_name]
+
+                        # If the host variable is conditionally allocated, create a pointer for it
+                        if not conditional == '.true.':
+                            # Reuse existing temporary pointer variable, if possible; otherwise add a local pointer (tmpptr)
```

Comment:
If it costs nothing to create a pointer, why bother reusing them?
Below I see that the local pointer is nullified after the scheme call, so there shouldn't be any conflicts in subsequent calls. Or am I misinterpreting the context all together?

Reply:
reusing refers to the SAME pointer being reused for the SAME variable in the different scheme calls. Otherwise, it would be creating one pointer N times for the same variable if it gets passed to N schemes in that group.